*** L0.95/swap.c	Mon Mar  9 11:39:02 1992
--- swap.c	Thu Mar 12 11:48:25 1992
***************
*** 159,170 ****
  
  repeat:
  	while (counter > 0) {
- 		counter -= 1024;
- 		dir_entry++;
  		if (dir_entry >= 1024)
  			dir_entry = FIRST_VM_PAGE>>10;
  		if (pg_table = pg_dir[dir_entry])
  			break;
  	}
  	if (counter <= 0) {
  		printk("Out of swap-memory\n");
--- 159,170 ----
  
  repeat:
  	while (counter > 0) {
  		if (dir_entry >= 1024)
  			dir_entry = FIRST_VM_PAGE>>10;
  		if (pg_table = pg_dir[dir_entry])
  			break;
+ 		counter -= 1024;
+ 		dir_entry++;
  	}
  	if (counter <= 0) {
  		printk("Out of swap-memory\n");
***************
*** 181,186 ****
--- 181,187 ----
  		page_entry++;
  		if (page_entry >= 1024) {
  			page_entry = -1;
+ 			dir_entry++;
  			goto repeat;
  		}
  		if (try_to_swap_out(page_entry + (unsigned long *) pg_table))