*** binutils-1.9/gprof.c	Sun Mar 17 05:36:43 1991
--- binutils/gprof.c	Sat Jun 13 13:25:53 1992
***************
*** 842,850 ****
      int	n;
      unsigned long pos;
      struct mesym *sym;
!     int incr = (hdr.high - hdr.low) / (nhist - 1);
  
!     if ((hdr.high - hdr.low) != 4 * (nhist - 1))
        abort ();
  
      for (n=0, pos=hdr.low, sym= &syms[0]; n<nhist; n++, pos+=incr) {
--- 842,850 ----
      int	n;
      unsigned long pos;
      struct mesym *sym;
!     int incr = (hdr.high - hdr.low) / (nhist /* - 1 */);
  
!     if ((hdr.high - hdr.low) != 4 * (nhist /* - 1 */))
        abort ();
  
      for (n=0, pos=hdr.low, sym= &syms[0]; n<nhist; n++, pos+=incr) {
***************
*** 2430,2437 ****
--- 2430,2439 ----
  {
    struct file_to_name *tail;
  
+ #ifdef _IOWRT
    if(stream->_flag & _IOWRT)
      fflush (stream);
+ #endif
    if (ferror (stream))
      fatal ("I/O error on `%s'", stream_name (stream));
    if (fclose (stream)==EOF)
