diff -rc2N poe-IGL/Makefile poe-new/Makefile 
*** poe-IGL/Makefile Sat Feb 22 23:12:03 1992 
--- poe-new/Makefile Mon Mar 2 06:07:35 1992 
*************** 
*** 3,16 **** 
  # All this code is freely distributable. 
  
! CC = gcc 
  CFLAGS = -O -I. 
  
! loginobj = login.o utmp2.o putenv.o setenv.o 
! whoobj = who.o utmp2.o 
  encryptobj = encrypt.o 
  lastobj = last.o 
! gettyobj = agetty.o utmp2.o 
  initobj = simpleinit.o 
! writeobj = write.o utmp2.o getlogin.o 
  
  .c.o: 
--- 3,16 ---- 
  # All this code is freely distributable. 
  
! CC = gcc -s 
  CFLAGS = -O -I. 
  
! loginobj = login.o 
! whoobj = who.o 
  encryptobj = encrypt.o 
  lastobj = last.o 
! gettyobj = agetty.o 
  initobj = simpleinit.o 
! writeobj = write.o 
  
  .c.o: 
*************** 
*** 17,25 **** 
        ${CC} ${CFLAGS} -c $*.c 
  
! all: libufc.a login who encrypt last agetty hostname init getlogin.o \ 
        write 
  
  install: all 
- strip login who encrypt last agetty hostname init write 
        cp login init agetty /bin 
        chmod 744 /bin/init /bin/agetty 
--- 17,24 ---- 
        ${CC} ${CFLAGS} -c $*.c 
  
! all: libufc.a login who encrypt last agetty hostname init \ 
        write 
  
  install: all 
        cp login init agetty /bin 
        chmod 744 /bin/init /bin/agetty 
*************** 
*** 65,68 **** 
        rm -f *.o 
  
! Clean: 
! rm -f *.o login who encrypt last agetty hostname init libufc.a write 
--- 64,67 ---- 
        rm -f *.o 
  
! Clean: clean 
! rm -f login who encrypt last agetty hostname init libufc.a write 
diff -rc2N poe-IGL/last.c poe-new/last.c 
*** poe-IGL/last.c Wed Feb 12 22:52:57 1992 
--- poe-new/last.c Sun Mar 1 14:04:09 1992 
*************** 
*** 8,11 **** 
--- 8,12 ---- 
  #include <time.h> 
  #include <sys/time.h> 
+ #include <unistd.h> 
  
  main(int argc, char *argv[]) 
*************** 
*** 22,29 **** 
        (void)lseek(fd, (off_t)sizeof(struct utmp), SEEK_END); 
  
! printf("Tty User Time\n"); 
        while((pos = lseek(fd, (off_t)-2*sizeof(struct utmp), SEEK_CUR)) >= 0) { 
                (void)read(fd, (char *)&ut, sizeof(struct utmp)); 
! printf("%2s %-8s %s", ut.ut_id, ut.ut_user, 
                        ctime(&ut.ut_time)); 
        } 
--- 23,30 ---- 
        (void)lseek(fd, (off_t)sizeof(struct utmp), SEEK_END); 
  
! printf("Tty User Time\n"); 
        while((pos = lseek(fd, (off_t)-2*sizeof(struct utmp), SEEK_CUR)) >= 0) { 
                (void)read(fd, (char *)&ut, sizeof(struct utmp)); 
! printf("%-12s %-8s %s", ut.ut_line, ut.ut_user, 
                        ctime(&ut.ut_time)); 
        } 
diff -rc2N poe-IGL/login.c poe-new/login.c 
*** poe-IGL/login.c Wed Feb 26 21:58:20 1992 
--- poe-new/login.c Sun Mar 1 14:02:10 1992 
*************** 
*** 84,88 **** 
   */ 
  
! #define TESTING 
  
  #ifdef TESTING 
--- 84,88 ---- 
   */ 
  
! #undef TESTING 
  
  #ifdef TESTING 
*************** 
*** 145,148 **** 
--- 145,149 ---- 
  #define TTYGRPNAME "other" 
  #endif 
+ #define MAXPATHLEN _POSIX_PATH_MAX 
  
  /* 
*************** 
*** 626,630 **** 
                struct stat st; 
  
! motd(); 
                (void)sprintf(tbuf, "%s/%s", _PATH_MAILDIR, pwd->pw_name); 
                if (stat(tbuf, &st) == 0 && st.st_size != 0) 
--- 627,631 ---- 
                struct stat st; 
  
! /* motd(); */ 
                (void)sprintf(tbuf, "%s/%s", _PATH_MAILDIR, pwd->pw_name); 
                if (stat(tbuf, &st) == 0 && st.st_size != 0) 
*************** 
*** 730,733 **** 
--- 731,735 ---- 
  #endif 
  
+ /* 
  jmp_buf motdinterrupt; 
  
*************** 
*** 752,755 **** 
--- 754,758 ---- 
        longjmp(motdinterrupt, 1); 
  } 
+ */ 
  
  checknologin() 
diff -rc2N poe-IGL/param.h poe-new/param.h 
*** poe-IGL/param.h Wed Feb 12 22:52:58 1992 
--- poe-new/param.h Thu Feb 27 08:54:45 1992 
*************** 
*** 6,10 **** 
  #define HZ 100 
  #define EXEC_PAGESIZE 4096 
- #define getdtablesize() (NR_OPEN-1) 
  
  #define NGROUPS 32 /* Max number of groups per user */ 
--- 6,9 ---- 
diff -rc2N poe-IGL/putenv.c poe-new/putenv.c 
*** poe-IGL/putenv.c Wed Feb 12 22:52:58 1992 
--- poe-new/putenv.c Sun Mar 1 12:36:35 1992 
*************** 
*** 16,20 **** 
  not, write to the Free Software Foundation, Inc., 675 Mass Ave, 
  Cambridge, MA 02139, USA. */ 
- #define USG 
  #include <sys/types.h> 
  #include <errno.h> 
--- 16,19 ---- 
*************** 
*** 41,46 **** 
  /* Put STRING, which is of the form "NAME=VALUE", in the environment. */ 
  int 
! putenv (string) 
! char *string; 
  { 
    char *name_end = index (string, '='); 
--- 40,44 ---- 
  /* Put STRING, which is of the form "NAME=VALUE", in the environment. */ 
  int 
! putenv (char *string) 
  { 
    char *name_end = index (string, '='); 
diff -rc2N poe-IGL/setenv.c poe-new/setenv.c 
*** poe-IGL/setenv.c Wed Feb 12 22:52:58 1992 
--- poe-new/setenv.c Sun Mar 1 13:01:02 1992 
*************** 
*** 7,16 **** 
  #include <memory.h> 
  #include <stdlib.h> 
  
  void *malloc(); 
  
! void setenv(var,val,overwrite) 
! char *var,*val; 
! int overwrite; 
    { char *p; 
      if(!overwrite && getenv(var)) 
--- 7,15 ---- 
  #include <memory.h> 
  #include <stdlib.h> 
+ #include <stdio.h> 
  
  void *malloc(); 
  
! void setenv(char *var, char *val, int overwrite) 
    { char *p; 
      if(!overwrite && getenv(var)) 
diff -rc2N poe-IGL/simpleinit.c poe-new/simpleinit.c 
*** poe-IGL/simpleinit.c Fri Feb 28 20:15:57 1992 
--- poe-new/simpleinit.c Sun Mar 1 14:05:43 1992 
*************** 
*** 12,16 **** 
  #include <sys/sysmacros.h> 
  
! #include "param.h" 
  #include "pathnames.h" 
  
--- 12,16 ---- 
  #include <sys/sysmacros.h> 
  
! #include <sys/param.h> 
  #include "pathnames.h" 
  
Binary files poe-IGL/ufc/crypt.o and poe-new/ufc/crypt.o differ 
Binary files poe-IGL/ufc/crypt_util.o and poe-new/ufc/crypt_util.o differ 
Binary files poe-IGL/ufc/ufc and poe-new/ufc/ufc differ 
diff -rc2N poe-IGL/utmp.h poe-new/utmp.h 
*** poe-IGL/utmp.h Tue Feb 25 21:40:58 1992 
--- poe-new/utmp.h Sun Mar 1 12:43:30 1992 
*************** 
*** 45,47 **** 
  void endutent(void); 
  
! #endif 
\ No newline at end of file 
--- 45,47 ---- 
  void endutent(void); 
  
! #endif 
diff -rc2N poe-IGL/utmp2.c poe-new/utmp2.c 
*** poe-IGL/utmp2.c Wed Feb 19 19:33:26 1992 
--- poe-new/utmp2.c Sun Mar 1 12:39:15 1992 
*************** 
*** 5,9 **** 
     conform to the XPG2 and SVID2 standards */ 
  
! #include "utmp.h" 
  #include <fcntl.h> 
  #include <unistd.h> 
--- 5,9 ---- 
     conform to the XPG2 and SVID2 standards */ 
  
! #include <utmp.h> 
  #include <fcntl.h> 
  #include <unistd.h> 
diff -rc2N poe-IGL/who.c poe-new/who.c 
*** poe-IGL/who.c Mon Feb 17 19:57:14 1992 
--- poe-new/who.c Sun Mar 1 12:55:37 1992 
*************** 
*** 13,20 **** 
        setutent(); 
  
! printf("User TTY Login-time\n"); 
        while(ut = getutent()) { 
                if(ut->ut_type == USER_PROCESS) 
! printf("%-8s %-2s %s", ut->ut_user, ut->ut_id, 
                                ctime(&ut->ut_time)); 
        } 
--- 13,20 ---- 
        setutent(); 
  
! printf("User tty Login-time\n"); 
        while(ut = getutent()) { 
                if(ut->ut_type == USER_PROCESS) 
! printf("%-8s %-12s %s", ut->ut_user, ut->ut_line, 
                                ctime(&ut->ut_time)); 
        } 
*************** 
*** 21,23 **** 
        endutent(); 
  } 
! 
\ No newline at end of file 
--- 21,23 ---- 
        endutent(); 
  } 
! 
*************** 
-- End of patches -- 

