--- tc.sig.h.org	Fri Nov 20 09:04:34 1992
+++ tc.sig.h	Sun Jan 10 11:12:56 1993
@@ -168,13 +168,18 @@
 /* Does not seem to work right... Christos */
 #  define SIGSYNCH       0 
 # endif
-# ifdef SIGSYNCH
-#  define SYNCHMASK 	(sigmask(SIGCHLD)|sigmask(SIGSYNCH))
-# else
-#  define SYNCHMASK 	(sigmask(SIGCHLD))
-# endif
-extern sigret_t synch_handler();
 #endif /* convex */
+
+#ifdef linux
+# define SIGSYNCH       SIGUSR1
+#endif
+
+#ifdef SIGSYNCH
+# define SYNCHMASK 	(sigmask(SIGCHLD)|sigmask(SIGSYNCH))
+extern sigret_t synch_handler();
+#else
+# define SYNCHMASK 	(sigmask(SIGCHLD))
+#endif
 
 #ifdef SAVESIGVEC
 # define NSIGSAVED 7
--- config/config.org	Sun Jan 10 11:14:20 1993
+++ config/config.linux	Sun Jan 10 11:14:31 1993
@@ -36,7 +36,7 @@
  *		Note that some machines eg. rs6000 have a vfork, but not
  *		with the berkeley semantics, so we cannot use it there either.
  */
-#define VFORK
+#undef VFORK
 
 /*
  * BSDJOBS	You have BSD-style job control (both process groups and
@@ -116,7 +116,6 @@
 #define DIRENT
 /****************** local defines *********************/
 
-/* Since people like shared libs so much, we'll make this a default. :-) */
 #define _PATH_TCSHELL "/bin/tcsh"
 #define ECHO_STYLE      BOTH_ECHO
 #ifdef BSDSIGS
