--- portmap-original/Makefile	Wed Aug 24 11:07:05 1994
+++ portmap/Makefile	Tue Aug 23 22:38:50 1994
@@ -36,7 +36,7 @@
 # With verbose logging on, HP-UX 9.x leaves zombies behind when SIGCHLD
 # is not ignored. Enable next macro for a fix.
 #
-# ZOMBIES = -DIGNORE_SIGCHLD
+ZOMBIES = -DIGNORE_SIGCHLD
 
 # Uncomment the following macro if your system does not have u_long.
 #
@@ -50,7 +50,7 @@
 
 # Auxiliary object files that may be missing from your C library.
 #
-AUX	= daemon.o strerror.o
+AUX	= daemon.o # strerror.o
 
 # Uncomment the following macro definitions if you are running SYSV.4.
 # This won't help much because SYSV.4 uses rpcbind instead of portmap.
@@ -61,15 +61,19 @@
 
 # Auxiliary libraries that you may have to specify
 #
-# LIBS	= -lrpc
+
+LIBS	= -lbsd # -lrpc
+
+LINUX = -D_USE_BSD -D_USE_BSD_SIGNAL
 
 ### End of configurable stuff.
 ##############################
 
 SHELL	= /bin/sh
 
-COPT	= -Dconst= -Dperror=xperror $(HOSTS_ACCESS) $(CHECK_PORT) \
-	$(SYS) -DFACILITY=$(FACILITY) $(ULONG) $(ZOMBIES) $(SA_LEN)
+COPT	= $(HOSTS_ACCESS) $(CHECK_PORT) \
+	$(SYS) -DFACILITY=$(FACILITY) $(ULONG) $(ZOMBIES) $(SA_LEN) $(LINUX)
+
 CFLAGS	= $(COPT) -O
 OBJECTS	= portmap.o pmap_check.o from_local.o $(AUX)
 
