*** /A/etc/termcap	Sun Dec  8 13:58:19 1991
--- termcap	Fri Dec 20 06:10:07 1991
***************
*** 52,54 ****
--- 52,85 ----
  	:it#8:ku=\E[A:kd=\E[B:kr=\E[C:kl=\E[D:kb=^H:\
  	:ho=\E[H:k1=\EOP:k2=\EOQ:k3=\EOR:k4=\EOS:pt:sr=\EM:vt#3:xn:\
  	:sc=\E7:rc=\E8:cs=\E[%i%d;%dr:
+ 
+ #
+ # Now - like console, but with function keys
+ #
+ # Standard key code definitions:
+ # ku,kd,kl,kr	sent by up-, down-, left-, right-arrow
+ # kb	    sent by backspace
+ # k1..k9,k0 sent by function key 1..10
+ # K1..5	    sent by keypad upper left,upper right,centre,lower left,lower right
+ # ka,kt,kB  sent by clear-all-tabs,clear-tab,backtab
+ # kT	    sent by set-tab-stop
+ # kC,kE,kS  sent by clear-screen,clear-to-eol,clear-to-eos
+ # kD,kL	    sent by delete-character,delete-line
+ # kh,kH	    sent by home,home-down
+ # kI,kA	    sent by insert char/enter-insert-mode,insert-line
+ # kM	    sent by exit-insert-mode
+ # kN,kP	    sent by next-page,previous-page
+ # kF,kR	    sent by scroll-forward,scroll-backwards
+ memacs:\
+ 	:do=^J:co#80:li#25:cl=\E[H\E[J:sf=\ED:\
+ 	:le=^H:bs:am:cm=\E[%i%d;%dH:nd=\E[C:up=\E[A:\
+ 	:ce=\E[K:cd=\E[J:so=\E[7m:se=\E[m:us=\E[4m:ue=\E[m:\
+ 	:md=\E[1m:mr=\E[7m:mb=\E[5m:me=\E[m:is=\E[1;25r\E[25;1H:\
+ 	:it#8:ku=\E[A:kd=\E[B:kr=\E[C:kl=\E[D:kb=^H:\
+ 	:ho=\E[H:kP=\E[5~:kN=\E[6~:kH=\E[Y:kh=\E[H:kD=\E[3~:kI=\E[2~:\
+ 	:k1=\E[[A:k2=\E[[B:k3=\E[[C:k4=\E[[D:k5=\E[[E:k6=\E[[F:\
+ 	:k7=\E[[G:k8=\E[[H:k9=\E[[I:k0=\E[[J:\
+ 	:K1=\E[[K:K2=\E[[L:\
+ 	:pt:sr=\EM:vt#3:xn:\
+ 	:sc=\E7:rc=\E8:cs=\E[%i%d;%dr:\
+ #
