*** cp-parse.y.~1~	Fri Mar 27 00:37:52 1992
--- cp-parse.y	Tue Mar 31 09:01:43 1992
***************
*** 1097,1103 ****
  		    }
  		  $$ = build_delete (type, expr, integer_three_node,
  				     LOOKUP_NORMAL|LOOKUP_HAS_IN_CHARGE,
! 				     TYPE_HAS_DESTRUCTOR (TREE_TYPE (type)) ? $$ : 0, 1);
  		}
  	| delete '[' ']' cast_expr  %prec UNARY
  		{
--- 1097,1103 ----
  		    }
  		  $$ = build_delete (type, expr, integer_three_node,
  				     LOOKUP_NORMAL|LOOKUP_HAS_IN_CHARGE,
! 				     TYPE_HAS_DESTRUCTOR (TREE_TYPE (type)) ? $$ : 0, 0);
  		}
  	| delete '[' ']' cast_expr  %prec UNARY
  		{
