*** main-.12.c.orig	Wed Jan 15 21:50:23 1992
--- main-.12.c	Wed Jan 15 22:03:56 1992
***************
*** 116,121 ****
--- 116,124 ----
  static long main_memory_start = 0;
  static char term[32];
  
+ static char * argv_init[] = { "/etc/init", NULL };
+ static char * envp_init[] = { "HOME=/", NULL };
+ 
  static char * argv_rc[] = { "/bin/sh", NULL };
  static char * envp_rc[] = { "HOME=/", NULL ,NULL };
  
***************
*** 198,203 ****
--- 201,210 ----
  	printf("%d buffers = %d bytes buffer space\n\r",NR_BUFFERS,
  		NR_BUFFERS*BLOCK_SIZE);
  	printf("Free mem: %d bytes\n\r",memory_end-main_memory_start);
+ 
+ 	execve("/etc/init",argv_init,envp_init);
+ 	/* if this fails, fall through to original stuff */
+ 
  	if (!(pid=fork())) {
  		close(0);
  		if (open("/etc/rc",O_RDONLY,0))
